### DISCUSSION TITLE
Nextjs build server doesn't see API port other than default

### DISCUSSION THREAD

**Discussion by mapawel** [vercel/next.js#37167] [Help]
Hi,

The fragment of my next.config.js containing URIs for server and for browser to shot for data to external API by Nextjs app:
```
serverRuntimeConfig: {
    URI: 'https://exampledomain.com:8443',
  },
  publicRuntimeConfig: {
    URI: 'https://exampledomain.com:8443',
  },
```
It results with no data fetch during build (when collecting page data...)

In this case "https://exampledomain.com:8443" means -> domain connected to digitalocean IP + a docker container on digitalocean Droplet with port exposed -> 8443 and HTTPS working Server (with SSL). This container is composed earlier before build. When I open in a browser "https://exampledomain.com:8443" before build, it works fine. Domain directs a request to digitalocean IP address and a port suits to container IP exposed external to the world sa a Nextjs server theoretically shoud see it.

When I try like this:
```
serverRuntimeConfig: {
    URI: 'https://exampledomain.com',
  },
  publicRuntimeConfig: {
    URI: 'https://exampledomain.com',
  },
```
without port and I expose port 443 (default port for HTTPS) from my API, it works! Nextjs during build sees the API. It means that as above a domain directs to the digitalocean IP. My API is exposed on port 443 so by default a domain shows to the world this API, even Nextjs server during build sees it!

But I want to have a container with Nextjs app on port 443 so my backend should be on other port (8443 for example). How to force Next js to fetch data during build from port 8443 or other than default 443?

Please, help.

**Comment by icyJoseph** [comment]
Mmm could this be related? https://github.com/node-fetch/node-fetch/issues/1488

Does it work at runtime? I cannot think of any other reason as to why this would happen, other than something strange going on the `node-fetch` library, that Next.js makes available as `fetch` (server side).

**Reply by mapawel** [reply]
Hi, thanks. I don't think it is connected to this issue. I use axios. My fetcher used for server in getStaticProps as well as for browser fetch (useSWR, data revalidation in browser) is:
```
export const fetchWebpage = async (url) => {
  const timer = startLoading();
  try {
    const response = await axios({
      url: `${API_URI}/api/webpage${url}`,
      method: 'GET',
      headers: {
        'content-type': 'application/json',
      },
    });
    stopLoading(timer);
    return response?.data;
  } catch (err) {
    stopLoading(timer);
    return err.response?.data;
  }
};
```
so depending on environment, variable API_URI gives me "proxy" address (for my example the same address but in my MacOS local environment there were two different localhosts) and it works perfect.

IMPORTANT! -> Few minutes ago I deployed again my backend (API) with no_default port -> https://exampledomain.com:50443.
I build the Nextjs App on VERCEL with the same settings and connection to https://exampledomain.com:50443 set as API_URI ,  AND I WORKS! Nextjs Server on vercel was able to fetch data from 50443. Maybe the problem is in Docker or in Digitalocean? I know it's getting wider problem than Nextjs however somebody could help how ti bite it? Please.
